### DISCUSSION TITLE
Trying to pass an access_token from the 1st step to the 2nd as an Authorization Bearer token

### DISCUSSION THREAD

**Discussion by vijaybalakrishnanintel** [postmanlabs/postman-flows#322] [Q&A]
I have the tests working in the Collections as a Runner with an access_token from the 1st step passed to the 2nd step using a variable access_token. 

But using it in the Flow, the access_token from the 1st step is not passed to the 2nd step. I pass it in a Test from the 1st step as an environment variable.

1. Have defined a variable called access_token as part of an Environment.
2. Test in 1st step:
pm.environment.set("access_token", access_token);
postman.setNextRequest("POSTjobsCognito");

3. Pre request script in 2nd Step:
const access_token = pm.environment.get("access_token");
console.log("POSTjobs access_token:", access_token);//doesn't show up here

I am also using Authorization Bearer with {{access_token}} from the 1st step.
Read this https://github.com/postmanlabs/postman-flows/discussions/142.
Tried using access_token as a Durable (since Assign Variables was not available) but not able to set a value from the response as data.body.access_token   ??

**Comment by productman-josh** [comment]
Don't rely much on Pre-Tests, updating Global/Environment variables, or Tests when using Postman Flows---it's really designed against that.

You'll need to keep the data stored in your Flow with Durables and Data. Almost everything you do should be point-and-click. 

I accomplished what you need in my Flow with:
1. Send Request
2. Connect Send Request's outgoing 'Response' to a 'Condition' block
3. Fill the Condition block with: /data/https/statusCode == 200
5. Connect Condition block's outgoing "Accept" block to a 'Create Durables' block
6. Add a 'token' variable to the 'Create Durables' block and set the value to: /data/body -- this will create @durable/token 
7. Connect the 'Create Durables' output to a new Send Request input
8. Your 'Send Request' requests...like, the actual selected request (not the Flows) should have something like {{token}} set in the header/body or somewhere so that you can use it in the Flow. If it does, then click the Send Request's  'Variables' option. You can select 'token' from the dropdown, then assigned it to @durable/token
8. Attach the output to the input of a terminal or whatever to continue working with the data. 

Everything should be done with drop-downs and such, letting the program show you what your options are most of the time.

**Reply by vellyr8x8** [reply]
Even i'm facing the same issue. When I send request to generate the access_token, the access_token is available at path data.body.access_token, but in durable block this path is not available.
<img width="828" alt="image" src="https://user-images.githubusercontent.com/94951386/172340783-2d56d9a5-7d1b-442a-9417-18e25138e08e.png">

**Comment by vijaybalakrishnanintel** [comment]
My response has data.body.access_token. I tried to use Condition(/data/http/statusCode == 200) --> Durable(Variable access_token = /data/body) but the response has it under data.body.access_token. So, it doesn't pick up the access_token variable.

**Reply by productman-josh** [reply]
Can you post an image of your flow? I think that would help communication

**Comment by vijaybalakrishnanintel** [comment]
<img width="1442" alt="postman flow" src="https://user-images.githubusercontent.com/32583555/171913124-03d26e3d-9597-48a7-8e8b-d39be473e7da.png">

**Comment by vijaybalakrishnanintel** [comment]
1. Get access token from Cognito Auth as data.body.access_token.Use it as a global var aka Durable
2. use access_token as req bearer token header in POstjobsCognito with req body.
3. Collect jobRequestId from response json and use it as a Durable(?)
4. Use jobRequestId and access_token to make PUTjobsCognito call
5.Use jobRequestId and access_token to make DELETEjobsCognito call

**Reply by productman-josh** [reply]
On the first block (auth request), that warning triangle indicates an error. Click the text directly to the left of the triangle to open the Flows Request. Post an image of that in here--- be sure to change any sensitive data but keep the same general format you are using. I believe you will likely need a durables block before that block, using it to assign some initial variables since collection/environment variables in the real request are largely ignored by Postman Flows

**Reply by samay-v** [reply]
Your first request doesn't have the examples defined, so the `Body` is not being recognised. Try [adding an example](https://learning.postman.com/docs/sending-requests/examples/) and then setting the durable
